### PR TITLE
Cleanup & Test framework

### PR DESCRIPTION
**Pull request by nishantjr**

### CODE REVIEW

**Review by msaxena2 — COMMENTED** [review]
I was thinking of getting rid of the original dependency on rostopic altogether, which would've required dependence on live* files. This is fine too, since it cleans up testing.

**Comment by nishantjr** [comment]
Okay. Any particular reason for getting rid of `rostopic`? Otherwise, did you approve? Can we merge this?

File: src/RVMaster/rvmonitor.rv [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+testSpecOne() {
+    int a = 10;
+    int b;
+
+    event controlMessageOne(std::string msg) /chatter std_msgs/String '{data:msg}'
+    {
+        msg = msg + "RV";
+        ROS_INFO("%s", msg.c_str());
+    }
+}
```
Comment (msaxena2) [inline src/RVMaster/rvmonitor.rv:14]: Should this file be placed under some sort of `test` folder to indicate it's only meant to be used as a test spec?

File: src/monitors/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+include_directories(
+  ${catkin_INCLUDE_DIRS}
+  include
+)
+
+catkin_package()
+
+# add_executable(monitor_node src/rvmonitor.cpp)
+# target_link_libraries(monitor_node ${catkin_LIBRARIES})
+
```
Comment (msaxena2) [inline src/monitors/CMakeLists.txt]: delete the commented out parts?

File: Test [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+progress "Building rosmop"
+(cd rosmop ; mvn -q package -DskipTests)
+
+progress "Building src"
+catkin_make -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTING=ON
+
+export ROS_MASTER_URI="http://localhost:11311/"
+export REAL_MASTER_URI="http://localhost:20022/"
+export ACCESS_POLICY_PATH=/home/njr/AA/ext/ROSRV/config/access-policy.cfg
```
Comment (msaxena2) [inline Test]: Is this path intentionally hardcoded?

File: Test [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    echo >&2 "$@"
+    exit $error_code
+}
+
+type pytest > /dev/null \
+    || clean_exit 1 -n \
+            'Please install pytest for python2 via pip.\n' \
+            'On ubuntu xenial run:: `pip2 install pip && pip2 install --user pytest`'
+
+progress() { echo ==== "$@" ; }
```
Comment (msaxena2) [inline Test]: line 5 uses progress. On my machine, running `./Test` throws `progress:Command not found`. Perhaps moving the definition above line 5 will fix this issue.

File: Test [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    error_code=$(shift)
+    echo >&2 "$@"
+    exit $error_code
+}
+
+type pytest > /dev/null \
+    || clean_exit 1 -n \
+            'Please install pytest for python2 via pip.\n' \
+            'On ubuntu xenial run:: `pip2 install pip && pip2 install --user pytest`'
+
```
Comment (msaxena2) [inline Test]: I simply got a `pytest command not found error` instead of the above mentioned message. Not sure why.

**Review by msaxena2 — CHANGES_REQUESTED** [review]
Looks good apart from the few minor issues.

**Comment by msaxena2** [comment]
> Okay. Any particular reason for getting rid of `rostopic`? Otherwise, did you approve? Can we merge this?

Rostopic isn't bundled with the minimal ROS installation, as far as I could tell. The only motivation was to depend on basic ROS.

**Comment by nishantjr** [comment]
Well, at the moment rvmaster cannot be built without it. Therefore it is
not test code. I will fix that in the next few PRs.

**Comment by nishantjr** [comment]
Ok. This is only for testing so should be fine.

**Comment by msaxena2** [comment]
lgtm.

**Review by msaxena2 — APPROVED** [review]

**Pull request merged**